vmtkScripts/contrib/vmtkmeshaddexternallayer.py

```diff
@@ -232,18 +232,15 @@ def Execute(self):
             arrayCalculator.SetResultArrayName('CalculatorResult')
             arrayCalculator.Update()
 
-            #This need to be copied in order to be of the right type (int)
-            cellEntityIdsArray.DeepCopy(arrayCalculator.GetOutput().GetCellData().GetArray('CalculatorResult'))
-
```

Comment:
Have you checked if the result is integer (preferably ID) type and has correct content?